File: workbench/poc_sd/src/main/java/Microservice.java
import org.json.simple.JSONObject;

import java.io.Serializable;

/**
 * Created by Simon Buchholz on 02.11.2016.
 */
public class Microservice implements Serializable {



    /** used as key in the p2p network */
    private String name;

    /** metadata */
    private String description;
    private String endpoint;

    private JSONObject swaggerJson;

    public Microservice(String name, String description, String endpoint) {
        this.name = name;
        this.description = description;
        this.endpoint = endpoint;
    }

    public Microservice(String name, String description, String endpoint, JSONObject swaggerJson) {
        this.name = name;
        this.description = description;
        this.endpoint = endpoint;
        this.swaggerJson = swaggerJson;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getEndpoint() {
        return endpoint;
    }

    public void setEndpoint(String endpoint) {
        this.endpoint = endpoint;
    }

    public JSONObject getSwaggerJson() {
        return swaggerJson;
    }

    @Override
    public String toString() {
        return "Microservice{" +
                "name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", endpoint='" + endpoint + '\'' +
                '}';
    }
}

File: workbench/poc_angular2_ipfs_ethereum/src/app/services/entities/microservice.ts
export class Microservice {

  private _name:string;
  private _description:string;
  private _hashToSwaggerFile:string;

  constructor(name: string, description: string, hashToSwaggerFile?:string) {
    this._name = name;
    this._description = description;
    this._hashToSwaggerFile = hashToSwaggerFile;
  }


  get name(): string {
    return this._name;
  }

  get description(): string {
    return this._description;
  }

  get hashToSwaggerFile(): string {
    return this._hashToSwaggerFile;
  }

  set name(name: string){
    this._name = name;
  }

  set description(description: string){
    this._description = description;
  }
  
  set hashToSwaggerFile(hashToSwaggerFile: string){
    this._hashToSwaggerFile = hashToSwaggerFile;
  }
}
